src/hoc/RealTime.tsx

```diff
@@ -0,0 +1,53 @@
+'use client'
+
+import { supabase } from '@/lib/supabase'
+import { selectTaskBoard, setTasks } from '@/redux/features/taskBoardSlice'
+import store from '@/redux/store'
+import { TaskResponse } from '@/types/dto/tasks.dto'
+import { RealtimePostgresChangesPayload } from '@supabase/supabase-js'
+import { usePathname } from 'next/navigation'
+import { ReactNode, useEffect } from 'react'
+import { useSelector } from 'react-redux'
+import { useRouter } from 'next/navigation'
+
+interface RealTimeTaskResponse extends TaskResponse {
+  deletedAt: string
+}
+
+export const RealTime = ({ children }: { children: ReactNode }) => {
+  const { tasks, token } = useSelector(selectTaskBoard)
+  const pathname = usePathname()
+  const router = useRouter()
+
+  const handleTaskRealTimeUpdates = (payload: RealtimePostgresChangesPayload<RealTimeTaskResponse>) => {
+    if (payload.eventType === 'INSERT') {
+      store.dispatch(setTasks([...tasks, payload.new]))
+    }
+    if (payload.eventType === 'UPDATE') {
+      const updatedTask = payload.new
+      if (updatedTask.deletedAt) {
+        const newTaskArr = tasks.filter((el) => el.id !== updatedTask.id)
+        store.dispatch(setTasks(newTaskArr))
+        if (pathname.includes('detail')) {
+          router.push(`/?token=${token}`)
+        }
+      } else {
+        const newTaskArr = [...tasks.filter((task) => task.id !== updatedTask.id), updatedTask]
+        store.dispatch(setTasks(newTaskArr))
+      }
+    }
+  }
+
+  useEffect(() => {
+    const channel = supabase
+      .channel('realtime tasks')
+      .on('postgres_changes', { event: '*', schema: 'public', table: 'Tasks' }, handleTaskRealTimeUpdates)
```

Comment:
interesting, how are we making sure that only the relevant tasks get updated in real time. E.g. a client should not get a real-time update for a task they don't have access to.

Reply:
It only shows the tasks that belongs to respective clients and respective workspace because it does filtering on the client side based on the workspace and clients.

Reply:
Can you confirm that filtering on client side based on workspace actually prevents this?

Let's say workspace A has tasks 1, 2, 3 and B has 4, 5, 6. Are we sure that changes to 1 / 2 / 3 wouldn't cause our channel to update data for workspace B?

Reply:
I have already confirmed it in the above comment. Adding a link to loom demo - 
[LOOM](https://www.loom.com/share/f13edbc5e18c49f683221ec92afd3c5d?sid=88eb2c27-a6bc-4146-b159-52ac67e7c438)